### ISSUE TITLE
wysiqyg-media subdirectories are created wrong when database as media storage is used

### ISSUE THREAD

**Issue by ludwig-gramberg** [magento/magento2#5868]
when using the database as media storage, magento checks wether or not the directories found in the database exist. unfortunately this checks goes wrong and it ends up creating folders just like the absolute path inside the wysiwyg folder.
### Preconditions
1.  magento 2.0.7
### Steps to reproduce
1. enable database media storage
2. bring up a cms page
3. click insert image
### Expected result
1.  no uncessary folders are being created
### Actual result
1. unnecessary folders are being created (and never used)

Example:
if magento resides in /var/www/html and thus you have the folder /var/www/html/pub/media/wysiwyg magento will create the folder /var/www/html/pub/media/wysiwyg/var/www/html/pub/media/wysiwyg

Afaik the Problem lies in vendor/magento/module-cms/Model/Wysiwyg/Images/Storage.php in the method createSubDirectories($path) on line 199-201 a directory is supposed to create a path, but is given an absolute path, but the directory creates this relative to its own location, thus the wring result in the path

**Comment by orlangur** [comment]
Does it cause any other problem than creating one redundant empty folder?

**Comment by ludwig-gramberg** [comment]
i just deleted my whole wysiwyg folder and now magento is unable to synch the files back from the database because the folders are missing, so its not just obsolete wrong folders but actually the ones magento needs are missing (what else would be the point of creating them in the first place)

tl;dr yes it breaks the media synch for cms/wysiwyg

**Comment by ludwig-gramberg** [comment]
my fix is this:
replace:
`$fullPath = rtrim($path, '/') . '/' . $directory['name'];
$this->_directory->create($fullPath);`
with:
`$this->_directory->create($directory['path'].'/'.$directory['name']);`

**Comment by davidbilly** [comment]
@ludwig-gramberg is it this issue? #5705

**Comment by ludwig-gramberg** [comment]
nope, this is something different

**Comment by ludwig-gramberg** [comment]
this appears to be fixed in 2.1.2 can someone form the magento team confirm this?

**Comment by IlnitskiyArtem** [comment]
@ludwig-gramberg Unfortunately I could not reproduce this issue.
I need more information about this issue in order to successfully investigate potential fixes. Please use our issue tracking template to format your description. Specify the exact steps you used.

**Comment by ludwig-gramberg** [comment]
@IlnitskiyArtem well I did, the faulty behaviour is that a folder that is intended as an absolut path is created in relation to the existing media folder, thus creating faulty paths like this one:
/var/www/html/pub/media/wysiwyg/var/www/html/pub/media/wysiwyg
it repeats the absolute path within the media directory

**Comment by ludwig-gramberg** [comment]
@IlnitskiyArtem also as I wrote I did not observe this behaviour any more since version 2.1.2

**Comment by veloraven** [comment]
I'm closing this issue as it is not reproduces any more.
If you can still reproduce this issue, please create a new GitHub issue report.
